src/app/api/slippage/[...slug]/route.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  getTokenByChainAndAddress,
+  getTokenByChainAndSymbol,
+} from '@indexcoop/tokenlists'
+import { headers } from 'next/headers'
+import { NextResponse } from 'next/server'
+import { isAddress, parseUnits } from 'viem'
+
+import { getQuote } from '@/app/api/quote/utils'
+import { fetchTokenMetrics } from '@/lib/utils/api/index-data-provider'
+
+import { formatWei } from '@/lib/utils'
+import type { NextRequest } from 'next/server'
+
+const slippageDefault = 0.5
+const slippageMax = 5.5
+
+export async function GET(
+  _: NextRequest,
+  { params }: { params: { slug: string[] } },
+) {
+  try {
+    const { slug } = params
+    const chainId = Number(slug[0])
+    const address = slug[1]
+
+    if (!address || !isAddress(address)) {
+      return NextResponse.json('Bad Request', { status: 400 })
+    }
+
+    const token = getTokenByChainAndAddress(chainId, address)
+    const usdc = getTokenByChainAndSymbol(chainId, 'USDC')
+
+    if (!token || !usdc) {
+      return NextResponse.json('Bad Request', { status: 400 })
+    }
+
+    const headersList = await headers()
+    const host = headersList.get('host')
+    const metrics = await fetchTokenMetrics({
+      hostname:
+        process.env.NODE_ENV === 'development'
+          ? `http://${host}`
+          : `https://${host}`,
+      chainId,
+      tokenAddress: address,
+      metrics: ['nav'],
+    })
+
+    const nav = metrics?.NetAssetValue
+    const inputAmount = nav === undefined ? '1000' : (nav * 1.1).toString()
+
+    if (!nav) {
+      return {
+        chainId,
+        address,
+        slippage: slippageDefault,
+      }
+    }
+
+    const quoteRequest: {
+      account: string
+      chainId: string
+      inputToken: string
+      outputToken: string
+      inputAmount: string
+      outputAmount: string
+      slippage: string
+    } = {
+      account: '0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045',
+      chainId: String(chainId),
+      inputToken: usdc.address,
+      outputToken: token.address,
+      // At the moment, we always wanna set both (inputAmount and indexTokenAmount)
+      inputAmount: parseUnits(inputAmount.toString(), usdc.decimals).toString(),
+      outputAmount: parseUnits('1', token.decimals).toString(),
+      slippage: String(0.5),
+    }
+
+    const quote = await getQuote(quoteRequest)
+
+    const quoteUsd = Number.parseFloat(
+      formatWei(quote.inputAmount, usdc.decimals),
+    )
+
+    const slippage = (quoteUsd - nav) / nav
+    const slippagePercentage = slippage * 100
+    const slippageOrMax =
+      slippagePercentage > slippageMax ? slippageMax : slippagePercentage
```

Comment:
Only return slippage if not greater than the max (5.5%)

Slippage is calculated by getting the diff between quote amount in USD and the NAV price. It's `quoteUsd - nav` to have a positive diff value most of the times. Negative values (excess) will be sorted out below in the response.